File: frontend/src/types/content-factory.ts
export const CONTENT_FACTORY_SECTIONS = [
  { id: "project-promotion", label: "项目推广", enabled: true },
  { id: "series", label: "系列推荐", enabled: false },
  { id: "weekly", label: "周榜导读", enabled: false },
] as const

export type ContentFactorySectionId = (typeof CONTENT_FACTORY_SECTIONS)[number]["id"]

export const RECOMMEND_PLATFORMS = [
  { id: "xiaohongshu", label: "小红书" },
  { id: "wechat", label: "公众号" },
  { id: "twitter", label: "Twitter / X" },
  { id: "linkedin", label: "LinkedIn" },
] as const

export type RecommendPlatform = (typeof RECOMMEND_PLATFORMS)[number]["id"]

export type ContentFactoryView = "source" | RecommendPlatform

export type PlatformVariant = {
  title?: string | null
  body?: string | null
  title_options?: string[]
  highlight_tags?: string[]
}

export const PROMOTION_STEPS = [
  { id: 1, label: "选择项目" },
  { id: 2, label: "项目分析" },
  { id: 3, label: "编辑优化" },
  { id: 4, label: "导出发布" },
] as const

export type CoverStyleSource = "builtin" | "manual" | "ai_generated"

export interface CoverStyleDesignAnalysis {
  design_category: string
  design_system: string
  typography_strategy: {
    title_ratio: string
    weight: string
    hierarchy_levels: string
  }
  layout_system: {
    structure: string
    alignment: string
  }
  color_strategy: {
    main_color: string
    accent_color: string
    background_note: string
  }
  information_density: string
  whitespace_usage: string
  visual_components: string[]
  overall_mood: string
  unique_memory_point: string
}

export interface ContentFactoryCoverStyle {
  id: string
  label: string
  source: CoverStyleSource
  prompt_prefix: string
  prompt_template: string
  negative_prompt: string
  color_tokens: { background: string; accent: string; text_safe: string }
  font_tokens: { heading: string; body: string; accent: string }
  style_report?: string | null
  design_analysis?: CoverStyleDesignAnalysis | null
  example_image_url?: string | null
  reference_image_url?: string | null
  fork_from_style_id?: string | null
  hidden?: boolean
  is_deletable?: boolean
  created_at?: string | null
}

export interface CoverStyleOption {
  id: string
  label: string
  source?: CoverStyleSource
  example_image_url?: string | null
}

export type CoverStyleGenerateStreamEvent =
  | { event: "start" }
  | { event: "delta"; text: string }
  | {
      event: "done"
      payload: {
        name: string
        prompt_prefix: string
        prompt_template: string
        negative_prompt: string
        color_tokens: ContentFactoryCoverStyle["color_tokens"]
        font_tokens: ContentFactoryCoverStyle["font_tokens"]
        style_report: string
        design_analysis?: CoverStyleDesignAnalysis | null
      }
    }
  | { event: "error"; detail: string }

export const NATIVE_README_TEMPLATE: CoverStyleOption = {
  id: "native-readme",
  label: "README 首屏",
}

export interface ContentFactoryProjectBrief {
  id: number
  name: string
  full_name: string
  description: string | null
  stars: number
  language: string | null
}

export interface CoverVariantRecord {
  cover_image_path: string
  cover_source?: "readme_capture" | "ai_generated"
  cover_style_id?: string
  cover_size_preset_id?: string
  cover_prompt_hash?: string | null
  cover_readme_sha?: string | null
  cover_generated_at?: string | null
}

export interface ContentFactoryCopyJson {
  title_options?: string[]
  body?: string
  hashtags?: string[]
  highlight_tags?: string[]
  hook?: string
  cover_texts?: string[]
  cta?: string
  image_template?: string | null
  source_title?: string | null
  source_body?: string | null
  platform_variants?: Partial<Record<RecommendPlatform, PlatformVariant>>
  cover_image_path?: string | null
  cover_readme_sha?: string | null
  cover_generated_at?: string | null
  cover_source?: "readme_capture" | "ai_generated" | null
  cover_style_id?: string | null
  cover_style_source?: "builtin" | "manual" | "ai_generated" | null
  cover_size_preset_id?: string | null
  cover_prompt_hash?: string | null
  /** 风格 + 画幅 → 封面记录（切换模板时各自保留） */
  cover_variants?: Record<string, CoverVariantRecord> | null
}

export interface UploadCoverResponse {
  cover_url: string
  draft: ContentFactoryDraft
  cached: boolean
}

export interface ContentFactoryDraft {
  id: number
  project_library_id: number
  project_id: number
  kind: string
  platform: RecommendPlatform
  step: number
  status: "draft" | "generated"
  title: string | null
  body: string | null
  body_json: ContentFactoryCopyJson | null
  created_at: string
  updated_at: string
  project: ContentFactoryProjectBrief
}

export interface GenerateCopyResponse {
  draft: ContentFactoryDraft | null
  generated_copy: ContentFactoryCopyJson
  preview_only: boolean
}

export interface OptimizeSelectionResponse {
  optimized_text: string
}

File: backend/app/schemas/content_factory.py
from __future__ import annotations



from datetime import datetime

from typing import Literal



from pydantic import BaseModel, ConfigDict, Field, model_validator

from app.services.cover_style_design_analysis import CoverStyleDesignAnalysis



ContentFactoryPlatform = Literal["xiaohongshu", "wechat", "twitter", "linkedin"]

ContentFactoryStatus = Literal["draft", "generated"]

ContentFactoryCoverTemplate = Literal[

    "native-readme",

    "minimal-tech",

    "black-gold",

    "code-style",

    "gradient",

    "geek",

]

CoverStyleSource = Literal["builtin", "manual", "ai_generated"]

FontTokenKind = Literal["sans", "serif", "mono", "display"]

CoverSource = Literal["readme_capture", "ai_generated"]





class FontTokensRead(BaseModel):

    heading: FontTokenKind = "sans"

    body: FontTokenKind = "sans"

    accent: FontTokenKind = "mono"





class ColorTokensRead(BaseModel):

    background: str = "#0a0a0a"

    accent: str = "#39ff14"

    text_safe: str = "#ffffff"





class CoverStyleRead(BaseModel):

    id: str

    label: str

    source: CoverStyleSource = "builtin"

    prompt_prefix: str = ""

    prompt_template: str = ""

    negative_prompt: str = ""

    color_tokens: ColorTokensRead = Field(default_factory=ColorTokensRead)

    font_tokens: FontTokensRead = Field(default_factory=FontTokensRead)

    style_report: str | None = None

    design_analysis: CoverStyleDesignAnalysis | None = None

    example_image_url: str | None = None

    reference_image_url: str | None = None

    fork_from_style_id: str | None = None

    hidden: bool = False

    is_deletable: bool = False

    created_at: datetime | None = None





class CoverStyleListResponse(BaseModel):

    items: list[CoverStyleRead] = Field(default_factory=list)





class CoverStyleWrite(BaseModel):

    style_id: str | None = Field(default=None, description="省略则自动生成")

    label: str = Field(min_length=1, max_length=128)

    prompt_prefix: str = Field(min_length=1)

    prompt_template: str = Field(min_length=1)

    negative_prompt: str = ""

    color_tokens: ColorTokensRead = Field(default_factory=ColorTokensRead)

    font_tokens: FontTokensRead = Field(default_factory=FontTokensRead)

    style_report: str | None = None

    fork_from_style_id: str | None = None





class CoverStylePatch(BaseModel):

    label: str | None = Field(default=None, max_length=128)

    prompt_prefix: str | None = None

    prompt_template: str | None = None

    negative_prompt: str | None = None

    color_tokens: ColorTokensRead | None = None

    font_tokens: FontTokensRead | None = None

    style_report: str | None = None

    design_analysis: CoverStyleDesignAnalysis | None = None

    hidden: bool | None = None





class CoverStyleGenerateRequest(BaseModel):

    generation_brief: str | None = Field(default=None, max_length=4000)

    reference_id: str | None = None

    fork_from_style_id: str | None = None

    generate_example: bool = True

    auto_save: bool = True

    @model_validator(mode="after")
    def validate_brief_or_reference(self) -> CoverStyleGenerateRequest:
        brief = (self.generation_brief or "").strip()
        ref_id = (self.reference_id or "").strip()
        if not brief and not ref_id:
            raise ValueError("generation_brief 与 reference_id 至少填写一项")
        return self




class CoverStyleReferenceUploadResponse(BaseModel):

    reference_id: str

    preview_url: str




class CoverStyleSaveParsedRequest(BaseModel):

    name: str = Field(min_length=1, max_length=64)

    prompt_prefix: str = Field(min_length=1)

    prompt_template: str = Field(min_length=1)

    negative_prompt: str = ""

    color_tokens: ColorTokensRead = Field(default_factory=ColorTokensRead)

    font_tokens: FontTokensRead = Field(default_factory=FontTokensRead)

    style_report: str | None = None

    design_analysis: CoverStyleDesignAnalysis | None = None

    reference_id: str | None = None

    fork_from_style_id: str | None = None

    generate_example: bool = True


class CoverStyleRefinePromptTemplateRequest(BaseModel):
    prompt_template: str = Field(min_length=1)
    instruction: str = Field(min_length=1, max_length=2000)
    prompt_prefix: str | None = None


class CoverStyleRefinePromptTemplateResponse(BaseModel):
    prompt_template: str


class CoverStyleRefineRequest(BaseModel):
    instruction: str = Field(min_length=1, max_length=2000)
    label: str | None = None
    design_analysis: CoverStyleDesignAnalysis | None = None
    prompt_prefix: str = Field(min_length=1)
    prompt_template: str = Field(min_length=1)
    negative_prompt: str = ""
    color_tokens: ColorTokensRead = Field(default_factory=ColorTokensRead)
    font_tokens: FontTokensRead = Field(default_factory=FontTokensRead)
    style_report: str | None = None


class CoverStyleRefineResponse(BaseModel):
    design_analysis: CoverStyleDesignAnalysis = Field(default_factory=CoverStyleDesignAnalysis)
    prompt_prefix: str
    prompt_template: str
    negative_prompt: str = ""
    color_tokens: ColorTokensRead = Field(default_factory=ColorTokensRead)
    font_tokens: FontTokensRead = Field(default_factory=FontTokensRead)
    style_report: str = ""


class CoverStylePreviewRequest(BaseModel):

    size_preset_id: str = Field(default="xiaohongshu-34")

    force: bool = False





class CoverStylePreviewResponse(BaseModel):

    example_image_url: str

    style_id: str





class CoverStyleForkRequest(BaseModel):

    label: str | None = Field(default=None, max_length=128)

    hidden_source: bool = False





class CoverPromptRecord(BaseModel):

    image_prompt: str = ""

    negative_prompt: str = ""

    prompt_hash: str = ""

    style_id: str = ""





class ContentFactoryProjectBrief(BaseModel):

    id: int

    name: str

    full_name: str

    description: str | None = None

    stars: int = 0

    language: str | None = None





class ContentFactoryPlatformVariant(BaseModel):

    title: str | None = None

    body: str | None = None

    title_options: list[str] = Field(default_factory=list)

    highlight_tags: list[str] = Field(default_factory=list)





class ContentFactoryCopyJson(BaseModel):

    title_options: list[str] = Field(default_factory=list)

    body: str = ""

    hashtags: list[str] = Field(default_factory=list)

    highlight_tags: list[str] = Field(default_factory=list)

    hook: str | None = None

    cover_texts: list[str] = Field(default_factory=list)

    cta: str | None = None

    image_template: str | None = None

    source_title: str | None = None

    source_body: str | None = None

    platform_variants: dict[str, ContentFactoryPlatformVariant] | None = None

    cover_image_path: str | None = None

    cover_readme_sha: str | None = None

    cover_generated_at: str | None = None

    cover_source: CoverSource | None = None

    cover_style_id: str | None = None

    cover_style_source: CoverStyleSource | None = None

    cover_prompt_record: CoverPromptRecord | dict | None = None

    cover_size_preset_id: str | None = None

    cover_prompt_hash: str | None = None

    cover_variants: dict[str, dict] | None = None





class ContentFactoryDraftCreate(BaseModel):

    project_id: int

    platform: ContentFactoryPlatform = "xiaohongshu"





class ContentFactoryDraftUpdate(BaseModel):

    platform: ContentFactoryPlatform | None = None

    step: int | None = Field(None, ge=1, le=4)

    title: str | None = None

    body: str | None = None

    body_json: ContentFactoryCopyJson | dict | None = None

    status: ContentFactoryStatus | None = None





class ContentFactoryDraftRead(BaseModel):

    model_config = ConfigDict(from_attributes=True)



    id: int

    project_library_id: int

    project_id: int

    kind: str

    platform: ContentFactoryPlatform

    step: int

    status: ContentFactoryStatus

    title: str | None

    body: str | None

    body_json: dict | None

    created_at: datetime

    updated_at: datetime

    project: ContentFactoryProjectBrief





class GenerateCopyRequest(BaseModel):

    preview_only: bool = False

    regenerate: bool = False

    platform: ContentFactoryPlatform | None = None

    from_source: bool = False





class GenerateCopyResponse(BaseModel):

    draft: ContentFactoryDraftRead | None = None

    generated_copy: ContentFactoryCopyJson

    preview_only: bool = False





class OptimizeSelectionRequest(BaseModel):

    selected_text: str = Field(min_length=1, max_length=2000)

    full_body: str | None = Field(default=None, max_length=10000)





class OptimizeSelectionResponse(BaseModel):

    optimized_text: str





class GenerateAiCoverRequest(BaseModel):

    style_id: str = Field(..., min_length=1)

    size_preset_id: str = Field(default="xiaohongshu-34")

    force: bool = False





class UploadCoverResponse(BaseModel):

    cover_url: str

    draft: ContentFactoryDraftRead

    cached: bool





class RevealCoverResponse(BaseModel):

    absolute_path: str

    directory: str

